config/routes.rb

```diff
@@ -1,3 +1,3 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get 'organization/public/:id', to: 'organizations#show'
```

Comment:
Tenemos unos helpers en rails que nos permiten crear rutas respetando un formato `restful`, en esta ruta debería se así:
`get 'organizations/:id/public'`
Prestando atención a que el nombre del recurso está expresado en plural(como es el caso de organizations).
Para hacer esta ruta, siguiendo la convenciones de rails se puede hacer de esta manera:
```suggestion
 resources :organizations, only: [] do
    get 'public', on: :member
  end
```
Que nos genera una ruta como esta, `public_organization GET  /organizations/:id/public(.:format) organizations#public`

Con lo cual nos dirige a una acción llamada `public` en el controlador. Te recomiendo esta lectura con respecto a las rutas en rails.
https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default

Reply:
Muchas gracias, cambios realizados